crates/solver/src/settlement_simulation.rs

```diff
@@ -91,12 +103,36 @@ pub async fn simulate_and_error_with_tenderly_link(
     network_id: &str,
     block: u64,
     simulation_gas_limit: u128,
+    access_list_estimator: Arc<dyn AccessListEstimating>,
 ) -> Vec<Result<()>> {
+    let settlements = settlements.collect::<Vec<_>>();
```

Comment:
I think this `collect()` can be dropped by using the `settlements` iterator directly a few lines down.

Reply:
Yup. Copy paste.